tools/bin/hawqregister

```diff
@@ -939,10 +1082,41 @@ def main(options, args):
 
     failure_handler = FailureHandler(conn)
     # register
-    ins = HawqRegister(options, args[0], utility_conn, conn, failure_handler)
+    if options.yml_config and ispartition(options.yml_config):
+        ins = HawqRegisterPartition(options, args[0], utility_conn, conn, failure_handler)
+    else:
+        ins = HawqRegister(options, args[0], utility_conn, conn, failure_handler)
+    ins.prepare()
     ins.register()
     conn.close()
 
+def test():
+    def connectdb(options):
+        '''
+        Trying to connect database, return a connection object.
+        If failed to connect, raise a pg.InternalError
+        '''
+        url = dbconn.DbURL(hostname=options.host, port=options.port,
+                           dbname=options.database, username=options.user)
+        logger.info('try to connect database %s:%s %s' % (url.pghost, url.pgport, url.pgdb))
+        utility_conn = pg.connect(dbname=url.pgdb, host=url.pghost, port=url.pgport,
+                                  user=url.pguser, passwd=url.pgpass, opt='-c gp_session_role=utility')
+        conn = pg.connect(dbname=url.pgdb, host=url.pghost, port=url.pgport,
+                          user=url.pguser, passwd=url.pgpass)
+        return utility_conn, conn
+
+    # connect db
+    try:
+        utility_conn, conn = connectdb(options)
```

Comment:
where is `options` define, your test function has 0-parameter. Have you ever tried your `test` function locally?